server/src/main/java/org/hyperskill/community/flashcards/card/CardService.java

```diff
@@ -0,0 +1,45 @@
+package org.hyperskill.community.flashcards.card;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hyperskill.community.flashcards.card.model.Card;
+import org.hyperskill.community.flashcards.category.CategoryService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Service;
+
+import java.util.Objects;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class CardService {
+    private static final int pageSize = 20;
+    private final MongoTemplate mongoTemplate;
+    private final CategoryService categoryService;
+
+    public Page<Card> getCardsByCategory(String categoryId, int page) {
+        Objects.requireNonNull(categoryId, "Category ID cannot be null");
+        if (page < 0) {
+            throw new IllegalArgumentException("Page number is out of bounds");
+        }
```

Comment:
we could have hibernate validate on negative page param

Reply:
Thank you for noticing, you're absolutely right.